src/expr/proof_node.cpp

```diff
@@ -104,7 +104,16 @@ void ProofNode::getFreeAssumptionsMap(
       // unbind its assumptions
       for (const Node& a : cur->d_args)
       {
-        currentScope.erase(a);
+        auto scopeCt = scopeDepth.find(a);
+        Assert(scopeCt != scopeDepth.end());
+        if (scopeCt->second == 1)
+        {
+          scopeDepth.erase(scopeCt);
+        }
+        else
+        {
+          scopeCt->second -= 1;
+        }
```

Comment:
You could refactor this to always decrement and check if 0, in which case you erase.

Reply:
Done in 26dbf02cd.